sdk/storage/Azure.Storage.Blobs.Batching/src/BlobBatch.cs

```diff
@@ -0,0 +1,294 @@
+        ///
+        /// For more information, see
+        /// <see href="https://docs.microsoft.com/rest/api/storageservices/delete-blob" />.
+        /// </summary>
+        /// <param name="containerName">
+        /// The name of the container containing the blob to delete.
+        /// </param>
+        /// <param name="blobName">
+        /// The name of the blob to delete.
+        /// </param>
+        /// <param name="deleteOptions">
+        /// Specifies options for deleting blob snapshots.
+        /// </param>
+        /// <param name="accessConditions">
+        /// Optional <see cref="BlobAccessConditions"/> to add conditions on
+        /// deleting this blob.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> on successfully deleting.  The response
+        /// cannot be used until the batch has been submitted with
+        /// <see cref="BlobBatchClient.SubmitBatchAsync"/>.
+        /// </returns>
+        public virtual Response DeleteBlob(
+            string containerName,
+            string blobName,
+            DeleteSnapshotsOption? deleteOptions = default,
+            BlobAccessConditions? accessConditions = default)
+        {
+            var blobUri = new BlobUriBuilder(_client.Uri)
+            {
+                BlobContainerName = containerName,
+                BlobName = blobName
+            };
+            return DeleteBlob(
+                blobUri.ToUri(),
+                deleteOptions,
+                accessConditions);
+        }
+
+        /// <summary>
+        /// The <see cref="DeleteBlob(Uri, DeleteSnapshotsOption?, BlobAccessConditions?)"/>
+        /// operation marks the specified blob or snapshot for deletion. The
+        /// blob is later deleted during garbage collection.
+        ///
+        /// Note that in order to delete a blob, you must delete all of its
+        /// snapshots. You can delete both at the same time using
+        /// <see cref="DeleteSnapshotsOption.Include"/>.
+        ///
+        /// For more information, see
+        /// <see href="https://docs.microsoft.com/rest/api/storageservices/delete-blob" />.
+        /// </summary>
+        /// <param name="blobUri">
+        /// The blob to delete's primary <see cref="Uri"/> endpoint.
+        /// </param>
+        /// <param name="deleteOptions">
+        /// Specifies options for deleting blob snapshots.
+        /// </param>
+        /// <param name="accessConditions">
+        /// Optional <see cref="BlobAccessConditions"/> to add conditions on
+        /// deleting this blob.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> on successfully deleting.  The response
+        /// cannot be used until the batch has been submitted with
+        /// <see cref="BlobBatchClient.SubmitBatchAsync"/>.
+        /// </returns>
+        public virtual Response DeleteBlob(
+            Uri blobUri,
+            DeleteSnapshotsOption? deleteOptions = default,
```

Comment:
Why nullable? Same question for BlobAccessConditions.

Reply:
Because `BlobBaseClient.Delete` is nullable and I wanted to the exact same API.  Any changes to that should be mirrored here.

Reply:
I guess this begs the question of why it is nullable in BlobBaseClient.Delete..

Reply:
That's outside the scope of this PR.  I think there's a separate bug for DeleteSnapshotOptions, but fundamentally this API should copy however that one evolves.

Reply:
Created #7974 to track the issue.